cmds/rm/rm.go

```diff
@@ -11,81 +11,166 @@
 //     -i: interactive mode
 //     -v: verbose mode
 //     -R: remove file hierarchies
-//     -r: equivalent to -R
+
 package main
 
 import (
 	"bufio"
+	"errors"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
 )
 
-var (
-	flags struct {
-		r bool
-		v bool
-		i bool
+type rmFlags struct {
+	recursive   bool
+	verbose     bool
+	interactive bool
+}
+
+func isEmpty(file string) (bool, error, []os.FileInfo) {
+	fileList, err := ioutil.ReadDir(file)
+	if err != nil {
+		//ignore this boolean return value
+		return false, err, fileList
 	}
-	cmd = "rm [-Rrvi] file..."
-)
+	if len(fileList) == 0 {
+		return true, nil, fileList
+	}
+	return false, nil, fileList
+}
 
-func init() {
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
+func interactivePrompt(input *bufio.Scanner, s string, args ...interface{}) (bool, error) {
```

Comment:
nit: Go likes short, concise, descriptive names. You could just call it `prompt`. (up to you)